src/renderer/port-forward/port-forward-dialog.tsx

```diff
@@ -84,43 +87,51 @@ export class PortForwardDialog extends Component<Props> {
     this.desiredPort = this.currentPort;
   };
 
-  onClose = () => {
+  onClose = async () => {
+    await (async () => dialogState.onClose())();
```

Comment:
Looks weird. What it's for? Maybe better to try one of these if it's kinda timing issue or so:

```   ts
    window.setTimeout(callback)
    window.requestIdleCallback(callback)
    process.nextTick(callback)
```

Reply:
I wasn't sure how to do the `onClose`, so I duplicated how the `ok` was implemented for the confirm-dialog (https://github.com/lensapp/lens/blob/4f75acf2b40b3efdae3c5b49de443dfa08459db0/src/renderer/components/confirm-dialog/confirm-dialog.tsx#L91)

I just wanted to make sure the port-forward UI would update to reflect the changes after the dialog closed (and not block anything). I overcomplicated the problem. I have simplified the implementation.